frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1804,11 +1808,15 @@ type mutableOutput struct {
 	llb.Output
 }
 
-func withShell(img dockerspec.DockerOCIImage, args []string) []string {
+func withShell(cmd instructions.Command, img dockerspec.DockerOCIImage, args []string, warn linter.LintWarnFunc) []string {
 	var shell []string
 	if len(img.Config.Shell) > 0 {
 		shell = append([]string{}, img.Config.Shell...)
 	} else {
+		if warn != nil {
+			msg := linter.RuleJSONArgsRequired.Format(cmd.Name())
```

Comment:
This doesn't look like a rule directly related to `withShell` helper, but related to how `ENTRYPOINT/CMD` work (or more how these properties will work in `docker run` afterward). I think it makes more sense if there is a

```
if PrependShell && len(img.Config.Shell) == 0 {
  report()
}
```

In `dispatchArg/Entrypoint`.